pkg/api/ldap_debug.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/ldap"
+	"github.com/grafana/grafana/pkg/services/multildap"
+)
+
+var (
+	getLDAPConfig = multildap.GetConfig
+	newLDAP       = multildap.New
+)
+
+// LDAPAttribute is a serializer for user attributes mapped from LDAP. Is meant to display both the serialized value and the LDAP key we received it from.
+type LDAPAttribute struct {
+	ConfigAttributeValue string `json:"cfgAttrValue"`
+	LDAPAttributeValue   string `json:"ldapValue"`
+}
+
+// RoleDTO is a serializer for mapped roles from LDAP
+type RoleDTO struct {
+	OrgId   int64           `json:"orgId"`
+	OrgRole models.RoleType `json:"orgRole"`
+	GroupDN string          `json:"groupDN"`
+}
+
+// TeamDTO is a serializer for mapped Teams from LDAP
+type TeamDTO struct {
+	GroupDN string `json:"groupDN"`
+	TeamId  int64  `json:"teamId"`
+}
+
+// LDAPUserDTO is a serializer for users mapped from LDAP
+type LDAPUserDTO struct {
+	Name           *LDAPAttribute `json:"name"`
+	Surname        *LDAPAttribute `json:"surname"`
+	Email          *LDAPAttribute `json:"email"`
+	Username       *LDAPAttribute `json:"login"`
+	IsGrafanaAdmin *bool          `json:"isGrafanaAdmin"`
+	IsDisabled     bool           `json:"isDisabled"`
+	OrgRoles       *[]RoleDTO     `json:"roles"`
+	Teams          *[]TeamDTO     `json:"teams"`
```

Comment:
This is only relevant for Enterprise and likely to change _very_ soon once I put the other PR for that. Please avoid putting too much emphasis on that right now.